HikVisionSource/src/main/java/io/vantiq/extsrc/HikVisionSource/HikVision.java

```diff
@@ -0,0 +1,1460 @@
+    /// the uploadfile methos.
+    /// ImageUtil . using preexisting interface which is well known in the class
+    /// heirarchy .
+    ///
+    public abstract class HikUploadHelper implements ResponseHandler {
+
+        @Override
+        public void onSuccess(Object body, okhttp3.Response response) {
+
+        }
+
+        @Override
+        public void onError(List<VantiqError> errors, okhttp3.Response response) {
+
+        }
+
+        @Override
+        public void onFailure(Throwable t) {
+        }
+
+    }
+
+    /**
+     * Procudure responsible for uploading images accpeted by notificatio from
+     * Camera .
+     * 
+     * @param notification
+     */
+    private void sendNotificationWithUpload(ThermalNotification notification) {
+        ImageUtil iu = new ImageUtil();
+        iu.vantiq = new Vantiq((String) config.get("vantiqServer"));
+        iu.vantiq.setAccessToken((String) config.get("authToken"));
+
+        uploadImage(notification, iu);
+        /*
+         * SH responseHandler = new SH();
+         * 
+         * if (!(notification.ImageName == null || notification.ImageName == "")) { File
+         * f = new File(String.format("%s/%s", DVRImageFolderPath
+         * ,notification.ImageName)); String target = String.format("%s/%s",
+         * VantiqDocumentPath ,notification.ImageName);
+         * iu.uploadToVantiq(f,vantiqResourcePath, target,responseHandler);
+         * 
+         * notification.ImageName = target; }
+         * 
+         * if (!(notification.ThermalImageName == null || notification.ThermalImageName
+         * == "")) { File f = new File(String.format("%s/%s", DVRImageFolderPath
+         * ,notification.ThermalImageName)); String target = String.format("%s/%s",
+         * VantiqDocumentPath ,notification.ThermalImageName); iu.uploadToVantiq(f,
+         * vantiqResourcePath,target);//,responseHandler); notification.ThermalImageName
+         * = target ; // notification.ThermalImageName; }
+         * 
+         * 
+         * 
+         * oClient.sendNotification(notification);
+         */
+    }
+
+    /**
+     * Hanlde all type oc communication probelm raised by the Camera
+     * 
+     * @param pAlarmer
+     * @param pAlarmInfo
+     * @param dwBufLen
+     * @param pUser
+     */
+    public void ProcessCommAlarm(HCNetSDK.NET_DVR_ALARMER pAlarmer, HCNetSDK.RECV_ALARM pAlarmInfo, int dwBufLen,
+            Pointer pUser) {
+        HCNetSDK.NET_DVR_ALARMINFO struAlarmInfo = new HCNetSDK.NET_DVR_ALARMINFO();
+
+        struAlarmInfo.write();
+        Pointer pInfo = struAlarmInfo.getPointer();
+        pInfo.write(0, pAlarmInfo.RecvBuffer, 0, struAlarmInfo.size());
+        struAlarmInfo.read();
+
+        String strIP = new String(pAlarmer.sDeviceIP, StandardCharsets.US_ASCII).trim();
+        String stringAlarm = "";
+        int i = 0;
+
+        switch (struAlarmInfo.dwAlarmType) {
+            case 0:
```

Comment:
constants for the numbers here would be good.  That way, folks will know what's going on